test/jdk/jdk/crac/ContextOrderTest.java

```diff
@@ -253,31 +251,22 @@ private MockResource(List<String> recorder, Priority priority, String id) {
             this.recorder = recorder;
             this.priority = priority;
             this.id = id;
-        }
 
-        @Override
-        public Priority getPriority() {
-            return priority;
+            if (priority != null) {
+                priority.getContext().register(this);
```

Comment:
Registering to global context explicitly and to local one implicitly is confusing; if the mock does not require the `priority` field just remove the constructor arg and register it directly as well.